### PR TITLE
Development

### PR DESCRIPTION
**Pull request by 21stCheran**

### CODE REVIEW

File: .gitignore
Comment (Blue0ni) [inline .gitignore:39]:
```suggestion
# Eclipse
.settings/

# Ignore Gradle GUI config
```

do `git rm .settings/` as well

File: src/main/java/com/twentyonec/ItemsLogger/ItemPlayer.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/ItemPlayer.java]: Bruh, move `savePlayer` and `loadInventory` to another class. Literally the 3rd time I've told you to do this.

File: src/main/java/com/twentyonec/ItemsLogger/ItemsLogger.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/ItemsLogger.java]: Strings

File: src/main/java/com/twentyonec/ItemsLogger/commands/MainCommand.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/commands/MainCommand.java]: One class per file excepting inner classes

File: src/main/java/com/twentyonec/ItemsLogger/commands/ReloadConfig.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/commands/ReloadConfig.java]: [Bukkit has built in colour codes](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/ChatColor.html#translateAlternateColorCodes(char,%20java.lang.String))

File: src/main/java/com/twentyonec/ItemsLogger/commands/SearchLogs.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/commands/SearchLogs.java]: See previous comment

File: src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java]: First argument of `LIMIT` is how many rows to skip. Second arg is how many rows to take
```suggestion
			sb.append(" LIMIT ?, ?");
			args.add(min);
			args.add(max - min);
```

**Review by Blue0ni — CHANGES_REQUESTED** [review]

File: src/main/java/com/twentyonec/ItemsLogger/commands/OpenItemLog.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/commands/OpenItemLog.java]:
```suggestion
				String inventoryName = itemPlayer.getCause() + " Inventory";
				final Inventory inv = Bukkit.createInventory(null, 54, inventoryName);
```

File: src/main/java/com/twentyonec/ItemsLogger/commands/OpenItemLog.java
Comment (stein-ein) [inline src/main/java/com/twentyonec/ItemsLogger/commands/OpenItemLog.java]: As a rule of thumb, avoid magic values. Any strings/numbers/whatever that are hardcoded directly into some function without any labeling/explanation (like variable name) are called magic values. They should almost always be replaced with a constant/enum which would then be referenced wherever appropriate. This does three things:

1. Makes the purpose of the value very clear. It's obvious what value `NO_PERMISSION_MESSAGE` holds.
2. Reduces repetition. Instead of hardcoding the same value (like no permission error) in multiple locations, you just do it once.
3. Makes modification easy. Whenever you want to change the message, you just change it that one place and you're done.

I suggest you create an enum `Messages` in which you could define all messages and have some helper methods like `formatMessage` etc... That's generally how you handle messages in any plugin/program.

**Review by stein-ein — CHANGES_REQUESTED** [review]

File: src/main/java/com/twentyonec/ItemsLogger/listeners/DeathSave.java
Comment (Blue0ni) [inline src/main/java/com/twentyonec/ItemsLogger/listeners/DeathSave.java]: Always use `this` for class variables
```suggestion
					this.storage.saveItemPlayer(itemPlayer);
```

**Review by Blue0ni — APPROVED** [review]

**Comment by Blue0ni** [comment]
Wait for Stein's approval to merge

**Review by stein-ein — APPROVED** [review]

**Pull request merged**
